target/linux/ipq40xx/image/generic.mk

```diff
@@ -276,7 +276,28 @@ define Device/avm_fritzbox-7530
 	SOC := qcom-ipq4019
 	DEVICE_PACKAGES := fritz-caldata fritz-tffs-nand
 endef
-TARGET_DEVICES += avm_fritzbox-7530
+
+define Device/avm_fritzbox-7530-toshiba
+	$(call Device/FitImageLzma)
+	DEVICE_VENDOR := AVM
+	DEVICE_MODEL := FRITZ!Box 7530 (Toshiba NAND)
+	DEVICE_ALT0_VENDOR := AVM
+	DEVICE_ALT0_MODEL := FRITZ!Box 7520
+	SOC := qcom-ipq4019
+	DEVICE_PACKAGES := fritz-caldata fritz-tffs-nand
+endef
+TARGET_DEVICES += avm_fritzbox-7530-toshiba
+
+define Device/avm_fritzbox-7530-macronix
+	$(call Device/FitImageLzma)
+	DEVICE_VENDOR := AVM
+	DEVICE_MODEL := FRITZ!Box 7530 (Macronix NAND)
+	DEVICE_ALT0_VENDOR := AVM
+	DEVICE_ALT0_MODEL := FRITZ!Box 7520
+	SOC := qcom-ipq4019
+	DEVICE_PACKAGES := fritz-caldata fritz-tffs-nand
```

Comment:
I think you should add a `SUPPORTED_DEVICES:=avm_fritzbox-7530` here to make it possible to upgrade old avm_fritzbox-7530 images to the macronix variant.